crypto/init.c

```diff
@@ -729,3 +680,70 @@ int OPENSSL_atexit(void (*handler)(void))
     return 1;
 }
 
+/*
+ * Library  destructor (a.k.a. _fini()). Code here is derived
+ * from stuff found in providers/fips/self_test.c
+ *
+ * We do kind of  best effort cleanup. If given platform/compiler
+ * does not support module destructors then its up to application
+ * to explicitly call OPENSSL_cleanup();
+ */
+#if defined(_WIN32) || defined(__CYGWIN__)
+# ifdef __CYGWIN__
+/* pick DLL_[PROCESS|THREAD]_[ATTACH|DETACH] definitions */
+#  include <windows.h>
+/*
+ * this has side-effect of _WIN32 getting defined, which otherwise is
+ * mutually exclusive with __CYGWIN__...
+ */
+# endif
+
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved);
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+    switch (fdwReason) {
+    case DLL_PROCESS_DETACH:
+        OPENSSL_cleanup();
+        break;
+    default:
+        break;
+    }
+    return TRUE;
+}
+
+#elif defined(__GNUC__) && !defined(_AIX)
+
+void __attribute__((destructor))
+call_cleanup(void)
```

Comment:
I think you need to either declare this as static, or prototype the function to silence the compiler warning

Reply:
It cannot be static as it is a destructor IMO. So yeah, please add a prototype.